shell/platform/darwin/macos/framework/Source/AccessibilityBridgeMacDelegate.h

```diff
@@ -0,0 +1,37 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_DARWIN_MACOS_FRAMEWORK_SOURCE_AccessibilityBridgeMacDelegate_H_
+#define FLUTTER_SHELL_PLATFORM_DARWIN_MACOS_FRAMEWORK_SOURCE_AccessibilityBridgeMacDelegate_H_
+
+#import <Cocoa/Cocoa.h>
+
+#import "flutter/shell/platform/common/cpp/accessibility_bridge.h"
+#import "flutter/shell/platform/darwin/macos/framework/Source/FlutterEngine_Internal.h"
+
+namespace flutter {
+
+class AccessibilityBridgeMacDelegate : public AccessibilityBridge::AccessibilityBridgeDelegate {
+ public:
+  AccessibilityBridgeMacDelegate(FlutterEngine* flutterEngine);
+
+  void OnAccessibilityEvent(ui::AXEventGenerator::TargetedEvent targeted_event) override;
+  void DispatchAccessibilityAction(AccessibilityNodeId target,
+                                   FlutterSemanticsAction action,
+                                   std::vector<uint8_t> data) override;
+  std::unique_ptr<FlutterPlatformNodeDelegate> CreateFlutterPlatformNodeDelegate() override;
+
+ private:
+  FlutterEngine* _flutterEngine;
```

Comment:
Added ac omment below about this too - this seems like it wants to be a `scoped_nsobject`, or possibly some kind of weak_ptr. I'm not sure if that's used in the desktop.

It's an FML thing, but we might want to just move it to the darwin common instead.

Reply:
I took a look at the file in mac embedding, it looks like all of them are using raw ptr.